src/components/item/settings/GeolocationModalButton.tsx

```diff
@@ -0,0 +1,156 @@
+import { useRef, useState } from 'react';
+
+import {
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  Stack,
+  TextField,
+} from '@mui/material';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+
+import { CountryForm } from '@graasp/map';
+import { DiscriminatedItem } from '@graasp/sdk';
+import { COMMON } from '@graasp/translations';
+import { Button } from '@graasp/ui';
+
+import { useBuilderTranslation, useCommonTranslation } from '@/config/i18n';
+import notifier from '@/config/notifier';
+import { hooks, mutations } from '@/config/queryClient';
+import { BUILDER } from '@/langs/constants';
+
+type Props = {
+  item: DiscriminatedItem;
+};
+
+export const GeolocationModalButton = ({ item }: Props): JSX.Element => {
+  const { t } = useBuilderTranslation();
+  const { t: commonT } = useCommonTranslation();
+  const [open, setOpen] = useState(false);
+  const { data: geoloc } = hooks.useItemGeolocation(item.id);
+  const { mutate: saveGeoloc } = mutations.usePutItemGeolocation();
+
+  const helperLabelRef = useRef<HTMLInputElement>(null);
+  const addressLabelRef = useRef<HTMLInputElement>(null);
+  const latRef = useRef<HTMLInputElement>(null);
+  const lngRef = useRef<HTMLInputElement>(null);
```

Comment:
Any reason why we use refs and not controlled inputs ?

Reply:
Isn't it best for performance? If I use controlled buttons I also need `useEffect` on values.